File: c++/-1_oop/5_prototypeFungsiClass/prototypeClassBedaFile/Mahasiswa.cpp
#include <iostream>
#include "Mahasiswa.h"

Mahasiswa::Mahasiswa(string nama, string jurusan, int nim){
    this->nama = nama;
    this->jurusan = jurusan;
    this->nim = nim;
}

void Mahasiswa::outData(){
    cout << this->nama << endl;
    cout << this->jurusan << endl;
    cout << this->nim << endl;
}

Mahasiswa::~Mahasiswa(){
    cout << "object " << this->nama << "telah release" << endl;
}

File: java/-1_oop/src/refrensiObject/Mahasiswa.java
package refrensiObject;

class Mahasiswa {
    String nama;
    String jurusan;
    int nim;

    Mahasiswa(String nama, String jurusan, int nim){
        this.nama = nama;
        this.jurusan = jurusan;
        this.nim = nim;
    }

    void outData(){
        System.out.println(this.nama);
        System.out.println(this.jurusan);
        System.out.println(this.nim);
    }
}
